cmp/grid/Grid.js

```diff
@@ -556,6 +563,13 @@ export class Grid extends Component {
         this.model.agGridModel.agApi.sizeColumnsToFit();
     };
 
+    // Catches column pinning by dragging
+    onColumnPinned = (ev) => {
```

Comment:
I think this should be handled a little differently, to be more consistent with how we handle column visibility/position/width changes. For other column changes via drag, we intentionally do not update our GridModel column state until the `onDragStopped` event is fired, which is the best way to determine when the state change has been "committed" when dragging. 

Then in the event handlers for those events we update the GridModel column state only if the event was not triggered by dragging (the logic for that varies, but I think `onColumnVisible` is a good parallel to how the `onColumnPinned` handler should work).

Essentially I think synchronizing the ag-Grid pinned state to our column state should work exactly the same as the column visibility, and go through the same `noteAgGridColumnStateChange` method.